Update (#68)
OpenVoiceOS/community-docs: docs/config_ovos_audio.md

## Audio Service

> The audio service is responsible for loading TTS and Audio plugins

Reviewer: ```suggestion
The audio service is responsible for loading TTS (Text-To-Speech) and Audio plugins.
```

Author: file not used for now

> All audio playback is handled by this service

Reviewer: ```suggestion
All audio playback is handled by this service.
```

### Native playback

> Usually playback is triggered by some originating bus message, eg `"recognizer_loop:utterance"`, this message contains metadata that is used to determine if playback should happen.

Reviewer: ```suggestion
Usually playback is triggered by some originating bus message, eg `"recognizer_loop:utterance"`. This message contains metadata that is used to determine if playback should happen.
```

Reviewer: This feels very technical for this section given the rewrite.

Author: not used file for now

> `message.context` may contain a source and destination, playback is only triggered if a message destination is a native_source or if missing (considered a broadcast).

Reviewer: ```suggestion
`message.context` may contain a source and destination. Playback is only triggered if a message destination is a `native_source` or if missing (considered a broadcast).
```

> This separation of native sources allows remote clients such as an android app to interact with OVOS without the actual device where ovos-core is running repeating all TTS and music playback out loud

Reviewer: ```suggestion
This separation of native sources allows remote clients such as an Android app to interact with OVOS without the actual device where `ovos-core` is running repeating all TTS and music playback out loud.
```

You can learn more about message targeting [here](https://jarbashivemind.github.io/HiveMind-community-docs/mycroft/)

By default, only utterances originating from the speech client and ovos cli are considered native

> for legacy reasons the names for ovos cli and speech client are `"debug_cli"` and `"audio"` respectively

Reviewer: ```suggestion
For legacy reasons the names for OVOS CLI and speech client are `"debug_cli"` and `"audio"` respectively.
```

### TTS

> Two TTS plugins may be loaded at once, if the primary plugin fails for some reason the second plugin will be used.

Reviewer: ```suggestion
Two TTS plugins may be loaded at once. If the primary plugin fails for some reason the second plugin will be used.
```

> This allows you to have a lower quality offline voice as fallback to account for internet outages, this ensures your device can always give you feedback

Reviewer: ```suggestion
This allows you to have a lower quality offline voice as fallback to account for internet outages, ensuring your device can always give you spoken feedback.
```

```javascript
"tts": {
    "pulse_duck": false,
    "module": "ovos-tts-plugin-mimic2",

>     "fallback_module": "ovos-tts-plugin-mimic"

Reviewer: Consider a more modern pair of examples, although these are probably the most illustrative.

},
```

### Audio

You can enable additional Audio plugins and define the native sources described above under the `"Audio"` section of `mycroft.conf`

ovos-core uses OCP natively for media playback, you can learn more about OCP [here](https://openvoiceos.github.io/community-docs/OCP)

> OCP will decide when to call the Audio service and what plugin to use, the main use case is for headless setups without a GUI

Reviewer: ```suggestion
OCP will decide when to call the Audio service and what plugin to use. The main use case is for headless setups without a GUI.
```

> NOTE: mycroft-core has a `"default-backend"` config option, in ovos-core this option has been deprecated and is always OCP.

Reviewer: ```suggestion
NOTE: `mycroft-core` has a `"default-backend"` config option, in `ovos-core` this option has been deprecated and is always OCP.
```

```javascript
"Audio": {
    "native_sources": ["debug_cli", "audio"],

    "backends": {
      "OCP": {
        "type": "ovos_common_play",
        "active": true
      },
      "simple": {
        "type": "ovos_audio_simple",
        "active": true
      },
      "vlc": {
        "type": "ovos_vlc",
        "active": true
      }
    }
},
```

> # Configuration

Reviewer 2: should be in a new file not ovos audio?

Author: I eliminated that file from the `mkdocs.yml` file.  It was too in depth for the casual user.  More configuration can be added as needed I think.  This one in general got split into the separate sections. Just didn't delete the file yet.  Still might use some of the references as it goes along.

- [Reading Configuration](#reading-configuration)
- [Configuring Configuration](#configuring-configuration)
    * [protected_keys](#protected-keys)
    * [disable_user_config](#disable-user-config)
    * [disable_remote_config](#disable-remote-config)
- [Meta Configuration](#meta-configuration)
    * [ovos.conf](#ovosconf)

> ## Reading Configuration

Reviewer: This whole section feels out of place in this particular file. It's also quite a bit more technical, since only a developer will be explicitly calling `Configuration()`.

`ovos_config.config.Configuration` is a singleton that loads a single config
object. The configuration files loaded are determined by `ovos.conf` as described below and can be in either json or

> yaml format.

Reviewer: ```suggestion
YAML format.
```

if `Configuration()` is called the following configs would be loaded in this
order:

- `{core-path}`/configuration/mycroft.conf
- `os.environ.get('MYCROFT_SYSTEM_CONFIG')` or /etc/mycroft/mycroft.conf
- `os.environ.get('MYCROFT_WEB_CACHE')` or `XDG_CONFIG_PATH`/neon/web_cache.json
- ~/.mycroft/mycroft.conf (Deprecated)
- `XDG_CONFIG_DIRS` + /mycroft/mycroft.conf
- /etc/xdg/mycroft/mycroft.conf
- `XDG_CONFIG_HOME` (default ~/.config) + /mycroft/mycroft.conf

When the configuration loader starts, it looks in these locations in this order, and loads ALL configurations. Keys that
exist in multiple configuration files will be overridden by the last file to contain the value. This process results in
a minimal amount being written for a specific device and user, without modifying default distribution files.

> ## Configuring Configuration

Reviewer: ```suggestion
## Configuring the Configuration
```

### protected_keys

A `"protected_keys"` configuration section may be added to a `Default` or `System` Config file
(default `/etc/mycroft/mycroft.conf`). This configuration section specifies
other configuration keys that may not be specified in `remote` or `user` configurations.
Keys may specify nested parameters with `.` to exclude specific keys within nested dictionaries.
An example config could be:

```json
{
  "protected_keys": {
    "remote": [
      "gui_websocket.host",
      "websocket.host"
    ],
    "user": [
      "gui_websocket.host"
    ]
  }
}
```

This example specifies that `config['gui_websocket']['host']` may be specified in user configuration, but not remote.
`config['websocket']['host']` may not be specified in user or remote config, so it will only consider default
and system configurations.

### disable_user_config

If this config parameter is set to True in `Default` or `System` configuration,
no user configurations will be loaded (no XDG configuration paths).

### disable_remote_config

If this config parameter is set to True in `Default` or `System` configuration,
the remote configuration (`web_cache.json`) will not be loaded.

## Meta Configuration

> while `mycroft.conf` configures the voice assistant, `ovos.conf` configures the library

Reviewer: ```suggestion
While `mycroft.conf` configures the voice assistant, `ovos.conf` configures the library.
```

Author: file is not used

> what this means is that `ovos.conf` decides what files are loaded by the `Configuration` class described above, as an end user or skill developer you should never have to worry about this

Reviewer: ```suggestion
What this means is that `ovos.conf` decides what files are loaded by the `Configuration` class described above. As an end user or skill developer you should never have to worry about this.
```

> all XDG paths across OpenVoiceOS packages build their paths taking `ovos.conf` into consideration

Reviewer: ```suggestion
All XDG paths across OpenVoiceOS packages build their paths taking `ovos.conf` into consideration.
```

> this feature is what allows downstream voice assistants such as neon-core to change their config files to `neon.yaml`

Reviewer: ```suggestion
This feature is what allows downstream voice assistants such as `neon-core` to change their config files to `neon.yaml`
```
